tests/Unit/PayPalUrlGeneratorTest.php

```diff
@@ -86,11 +101,58 @@ private function newIncompletePayPalUrlConfig() {
 		return PayPalUrlConfig::newFromConfig( [
 			'base-url' => self::BASE_URL,
 			'account-address' => 'some@email-adress.com',
-			'notify-url' => 'http://my.donation.app/handler/paypal/',
-			'return-url' => 'http://my.donation.app/donation/confirm/',
-			'cancel-url' => 'http://my.donation.app/donation/cancel/',
+			'notify-url' => self::NOTIFY_URL,
+			'return-url' => self::RETURN_URL,
+			'cancel-url' => self::CANCEL_URL,
 			'item-name' => ''
 		] );
 	}
 
+	public function testDelayedSubscriptions() {
+		$generator = new PayPalUrlGenerator( $this->newPayPalUrlConfigWithDelayedPayment() );
+
+		$this->assertUrlValidForDelayedSubscriptions(
+			$generator->generateUrl( 1234, Euro::newFromString( '12.34' ), 3, 'utoken', 'atoken' )
+		);
+	}
+
+	private function assertUrlValidForDelayedSubscriptions( $generatedUrl ) {
+		$this->assertContains( 'https://www.sandbox.paypal.com/cgi-bin/webscr', $generatedUrl );
+		$this->assertContains( 'cmd=_xclick-subscriptions', $generatedUrl );
+		$this->assertContains( 'no_shipping=1', $generatedUrl );
+		$this->assertContains( 'src=1', $generatedUrl );
+		$this->assertContains( 'sra=1', $generatedUrl );
+		$this->assertContains( 'srt=0', $generatedUrl );
+		$this->assertContains( 'a1=0', $generatedUrl );
+		$this->assertContains( 'p1=90', $generatedUrl );
+		$this->assertContains( 't1=D', $generatedUrl );
+		$this->assertContains( 'a3=12.34', $generatedUrl );
+		$this->assertContains( 'p3=3', $generatedUrl );
+		$this->assertContains( 't3=M', $generatedUrl );
```

Comment:
A bunch of this stuff is repeated in other tests. Also, it's not apparent what the purpose of all these things is. Are all of these things critical to us, or are some of them things that just normally are in there? Are there logical groups that can be extracted into something like `assertContainsPaymentIntervalParameters`?